test/services/return-logs/fetch-download-return-log.service.test.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+const Sinon = require('sinon')
+
+const { describe, it, beforeEach, afterEach } = (exports.lab = Lab.script())
+const { expect } = Code
+
+// Test helpers
+const ReturnLogHelper = require('../../support/helpers/return-log.helper.js')
+const ReturnSubmissionHelper = require('../../support/helpers/return-submission.helper.js')
+const ReturnSubmissionModel = require('../../../app/models/return-submission.model.js')
+const ReturnSubmissionLineHelper = require('../../support/helpers/return-submission-line.helper.js')
+
+// Thing under test
+const FetchDownloadReturnLogService = require('../../../app/services/return-logs/fetch-download-return-log.service.js')
+
+describe('Fetch Download Return Log service', () => {
+  let returnLog
+  let returnSubmissions = []
+
+  beforeEach(async () => {
+    returnLog = await ReturnLogHelper.add()
+
+    // We stub on the model prototype so that any created instances have $applyReadings stubbed. We don't set any return
+    // value as we don't need it to actually do anything; we just want to be able to assert that it was called.
+    Sinon.stub(ReturnSubmissionModel.prototype, '$applyReadings')
+
+    returnSubmissions = await Promise.all([
+      ReturnSubmissionHelper.add({ returnLogId: returnLog.id, version: 1 }),
+      ReturnSubmissionHelper.add({ returnLogId: returnLog.id, version: 2 }),
+      ReturnSubmissionHelper.add({ returnLogId: returnLog.id, version: 3 })
+    ])
+
+    await Promise.all([
+      ReturnSubmissionLineHelper.add({
+        returnSubmissionId: returnSubmissions[0].id,
+        startDate: '2023-03-01',
+        quantity: 10
+      }),
+      ReturnSubmissionLineHelper.add({
+        returnSubmissionId: returnSubmissions[0].id,
+        startDate: '2023-01-01',
+        quantity: 5
+      })
+    ])
+  })
+
+  afterEach(() => {
+    Sinon.restore()
+  })
```

Comment:
Because these calls are in a `beforeEach()`, and we have 3 unit tests, it means we are creating the following every time this module is run.

- 3 return logs
- 9 return submissions
- 18 return submission lines

Whereas, if they were moved into a `before()` I reckon you could get away with just the following and still the tests would pass.

- 1 return log
- 3 return submissions
- 6 return submission lines

You may need both a `before()` and a `beforeEach()` because of the stubbing. But I reckon it's worth a go!